src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -141,37 +335,24 @@ void CoreWorkerDirectTaskSubmitter::OnWorkerIdle(
   // Return the worker if there was an error executing the previous task,
   // the previous task is an actor creation task,
   // there are no more applicable queued tasks, or the lease is expired.
-  if (was_error || current_queue.empty() ||
-      current_time_ms() > lease_entry.lease_expiration_time) {
+  if (!lease_entry.WorkerIsStealing() &&
```

Comment:
When would we call OnWorkerIdle for a worker that's stealing?

Reply:
Currently, there are two circumstances that lead to `OnWorkerIdle` being called on a thief while it is stealing:

1. Because the stolen tasks are forwarded to the thief individually (by the victim's `PushNormalTask` callbacks), a stolen task can complete its execution on the thief worker node (and thus respond to the owner by activating the thief's `PushNormalTask` callback, which in turn calls `OnWorkerIdle` on the thief) before all stolen tasks have been forwarded to the thief.
2. If new tasks are added to the task queue while stealing is in progress, `SubmitTask` will call `OnWorkerIdle` to submit the new tasks to a worker, and this worker can be the thief itself. In fact, I've seen this happen before. This scenario (tasks being added to the queue after work stealing has begun) is possible because sometimes there are some lags in the process of filling up the task queue. When one such lag is long enough, the queue can become empty and a worker can finish its load (thus entering the work stealing mode) before more tasks are added to the queue.